crypto/crypto.go

```diff
@@ -19,3 +33,68 @@ func SecureToken() string {
 func removePadding(token string) string {
 	return strings.TrimRight(token, "=")
 }
+
+// Generate key used to encrypt totp secret
+func getEncryptionKey() string {
+	passphrase := &passphrase{}
+	err := envconfig.Process("gotrue", passphrase)
+	if err != nil {
+		panic(fmt.Errorf("GOTRUE_PASSPHRASE not found: %v", err))
+	}
+	hasher := md5.New()
+	hasher.Write([]byte(passphrase.Value))
+	return hex.EncodeToString(hasher.Sum(nil))
+}
```

Comment:
The intuition behind this is to never store the passphrase in the config struct and make it a private function in the crypto package so that it can only be accessed by other functions in the same package.

Reply:
after some thought I think it might be better to use the raw passphrase to encrypt with, it improves compatibility if people want to integrate other systems that need the ability to encrypt/decrypt database secrets without having to know that gotrue has it's own encryption scheme (`AES(hash(passphrase),secret)`), and they can just use plain `AES(passphrase,secret)`

it also makes it more secure in that the pass phrase (I think?) can be variable length rather than a fixed length hash, so will be waaaaaaay harder to brute force (md5 alone would be relatively "easy" given enough time+compute)

Reply:
@awalias Ah I see where you're coming from. Actually, the original intention of using a md5 hash and converting it to hex is so that a user can provide a variable length passphrase and it will always be converted to a 32 byte key. If we don't use any hashing algorithm, then the user would have to ensure that the passphrase is 16, 24 or 32 bytes long or AES will throw an error.

Reply:
ahh ok, then we should enforce a 32 byte passphrase - we also want to strongly recommend that people generate this **randomly** and we should do this for the hosted version 👍